platform/broadcom/saibcm-modules/systems/bde/linux/user/kernel/linux-user-bde.c

```diff
@@ -1121,17 +1127,27 @@ _instance_attach(unsigned int inst_id, unsigned int dma_size)
     /* Reprobe the system for hot-plugged device */
     _device_reprobe();
 
+    gprintk("INFO: Request to attach to instance_id %d with dma size %d!\n", inst_id, dma_size);
+
+    spin_lock(&bde_resource_lock);
+
     /* Validate the resource with inst_id */
     exist = _instance_validate(inst_id, dma_size);
+
     if (exist < 0) {
+        spin_unlock(&bde_resource_lock);
         return LUBDE_FAIL;
     }
     if (exist > 0) {
+        gprintk("INFO: Already attached to instance_id %d with dma size %d!\n", inst_id, dma_size);
+        spin_unlock(&bde_resource_lock);
         return LUBDE_SUCCESS;
     }
     if (_dma_resource_alloc(dma_size, &dma_offset) < 0) {
+        spin_unlock(&bde_resource_lock);
```

Comment:
we should add gprintk here also for debugging

Reply:
Added printk for error cases.